File: algorithms/Sorting/Easy/SelectionSort/selectionSort.java


public class selectionSort {
    static void swap(int[] array, int i, int j) {
        int temp = array[i];
        array[i] = array[i];
        array[j] = temp;
    }

    static int findmin(int[] array, int i, int j) {
        int m = i;
        for (int k = i; k < j; j++) {
            if (array[m] < array[k]) {
                m = k;
            }
        }
        return m;
    }

    static void VSDselectionSort(int[] array, int n) {
        for (int i = 0; i < n; i++) {
            int j = findmin(array, i, n);
            swap(array, i, j);
        }
        for (int a : array) {
            System.out.print(a + " ");
        }
    }

    public static void main(String args[]) {
        int[] arr = { 1, 2, 53, 12, 45, 11, 56 };
        int n = arr.length;
        VSDselectionSort(arr, n);

    }

}

File: algorithms/Sorting/Easy/SelectionSort/selectionSort.py



def findmin(arr,left,right):
    m=left
    for i in range(left,right):
        if arr[i]<arr[left]:
            m=i 
    return m 
def selectionSort(arr):
    for i in range(len(arr)):
        j=findmin(arr,i,len(arr))
        arr[i],arr[j]=arr[j],arr[i]
    print(*arr)


if __name__=="__main__":
    arr=[1,2,53,12,45,11,56]
    selectionSort(arr)

